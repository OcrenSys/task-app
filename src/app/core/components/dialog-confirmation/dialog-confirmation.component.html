<mat-card class="card">
  <mat-card-header class="card-header">
    <mat-card-title>Eliminar tarea</mat-card-title>
  </mat-card-header>

  <mat-card-content>
    <p>Â¿Deseas eliminar esta tarea permanentemente?</p>
  </mat-card-content>

  <mat-card-actions class="mt-2">
    <button
      class="mh-2"
      mat-raised-button
      aria-label="cancel-action-buttom"
      (click)="onCancel()"
    >
      Cancelar
    </button>

    <button
      class="mh-2"
      color="warn"
      mat-raised-button
      aria-label="delete-action-buttom"
      (click)="onDelete()"
    >
      <mat-icon>delete</mat-icon>
      Eliminar
    </button>
  </mat-card-actions>
</mat-card>
