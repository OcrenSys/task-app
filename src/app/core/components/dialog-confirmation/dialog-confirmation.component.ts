import { Component } from '@angular/core';

import { MatButtonModule } from '@angular/material/button';
import { MatDividerModule } from '@angular/material/divider';
import { MatCardModule } from '@angular/material/card';
import { MatDialogRef } from '@angular/material/dialog';
import { MatIconModule } from '@angular/material/icon';

@Component({
  selector: 'app-dialog-confirmation',
  templateUrl: './dialog-confirmation.component.html',
  styleUrls: ['./dialog-confirmation.component.css'],
  standalone: true,
  imports: [MatCardModule, MatDividerModule, MatButtonModule, MatIconModule],
})
export class DialogConfirmationComponent {
  constructor(private dialogRef: MatDialogRef<DialogConfirmationComponent>) {}

  protected onCancel(): void {
    this.dialogRef.close({ confirmed: false });
  }

  protected onDelete(): void {
    this.dialogRef.close({ confirmed: true });
  }
}
