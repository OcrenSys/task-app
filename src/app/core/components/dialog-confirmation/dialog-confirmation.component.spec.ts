import { ComponentFixture, TestBed } from '@angular/core/testing';

import { DialogConfirmationComponent } from './dialog-confirmation.component';
import { MatButtonModule } from '@angular/material/button';
import { MatCardModule } from '@angular/material/card';
import { MatDividerModule } from '@angular/material/divider';
import { MatIconModule } from '@angular/material/icon';
import { MatDialogRef } from '@angular/material/dialog';

describe('DialogConfirmationComponent', () => {
  let component: DialogConfirmationComponent;
  let fixture: ComponentFixture<DialogConfirmationComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        DialogConfirmationComponent,
        MatCardModule,
        MatDividerModule,
        MatButtonModule,
        MatIconModule,
      ],
      providers: [
        {
          provide: MatDialogRef,
          useValue: {},
        },
      ],
    });
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(DialogConfirmationComponent);
    component = fixture?.componentInstance;
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('should match with snapshot', () => {
    const compile = fixture?.nativeElement as HTMLElement;
    expect(compile).toMatchSnapshot();
  });
});
