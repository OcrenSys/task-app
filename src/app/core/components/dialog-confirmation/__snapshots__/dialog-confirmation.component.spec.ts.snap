// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogConfirmationComponent should match with snapshot 1`] = `
<div
  id="root1"
>
  <mat-card
    class="mat-mdc-card mdc-card card"
  >
    <mat-card-header
      class="mat-mdc-card-header card-header"
    >
      <div
        class="mat-mdc-card-header-text"
      >
        <mat-card-title
          class="mat-mdc-card-title"
        >
          Eliminar tarea
        </mat-card-title>
      </div>
    </mat-card-header>
    <mat-card-content
      class="mat-mdc-card-content"
    >
      <p>
        Â¿Deseas eliminar esta tarea permanentemente?
      </p>
    </mat-card-content>
    <mat-card-actions
      class="mat-mdc-card-actions mdc-card__actions mt-2"
    >
      <button
        aria-label="cancel-action-buttom"
        class="mh-2 mdc-button mdc-button--raised mat-mdc-raised-button"
        mat-raised-button=""
        mat-ripple-loader-class-name="mat-mdc-button-ripple"
        mat-ripple-loader-uninitialized=""
      >
        <span
          class="mat-mdc-button-persistent-ripple"
        />
        <span
          class="mdc-button__label"
        >
           Cancelar 
        </span>
        <span
          class="mat-mdc-focus-indicator"
        />
        <span
          class="mat-mdc-button-touch-target"
        />
      </button>
      <button
        aria-label="delete-action-buttom"
        class="mh-2 mdc-button mdc-button--raised mat-mdc-raised-button mat-warn"
        color="warn"
        mat-raised-button=""
        mat-ripple-loader-class-name="mat-mdc-button-ripple"
        mat-ripple-loader-uninitialized=""
      >
        <span
          class="mat-mdc-button-persistent-ripple"
        />
        <mat-icon
          aria-hidden="true"
          class="mat-icon notranslate"
          role="img"
        >
          delete
        </mat-icon>
        <span
          class="mdc-button__label"
        >
           Eliminar 
        </span>
        <span
          class="mat-mdc-focus-indicator"
        />
        <span
          class="mat-mdc-button-touch-target"
        />
      </button>
    </mat-card-actions>
  </mat-card>
</div>
`;
